### ISSUE TITLE
[Feature] Add golden value for metric logging in ESM2 finetuning

### ISSUE THREAD

**Issue by sichu2023** [NVIDIA/bionemo-framework#684]
### Problem & Motivation

- current test on metric logging only test if metric is logged
- we need a golden value test on logged metric for ESM2 finetuning, where we overfit to a sample dataset

### BioNeMo Framework Version

4d61dc667e0053706fb0d7e5afe9737ae3a0526c on sichu/metric-config

### Category

Other

### Proposed Solution

- add golden values to `sub-packages/bionemo-esm2/tests/bionemo/esm2/scripts/test_finetune_esm2.py`

### Expected Benefits

- golden value test for finetuning and metric logging

### Code Example

```python
Related to PR https://github.com/NVIDIA/bionemo-framework/pull/677
```

**Comment by trvachov** [comment]
Closing as stale — this referenced ESM2 dataset paths in the `bionemo-esm2` sub-package. That sub-package has been completely removed and ESM2 was rewritten in `bionemo-recipes/models/`.

**Comment by trvachov** [comment]
Reopening — the previous closure comment was inaccurate. While the `bionemo-esm2` sub-package has been removed and ESM2 was rewritten in `bionemo-recipes/models/esm2/`, the underlying feature request is still valid:

**Current state of golden value test coverage for ESM2:**
- ✅ Pretraining/inference: `test_golden_values()`, `test_golden_values_thd()`, `test_golden_values_thd_padded()` compare TE vs HF reference outputs (loss, logits)
- ❌ Finetuning: `test_peft.py` only checks model creation and that loss is non-null — no golden value assertions on finetuning metrics

The request to add golden value tests for ESM2 finetuning metrics (e.g., overfitting to a sample dataset and asserting expected loss/metric values) remains unaddressed in the new codebase. Leaving open.
